Concurrent image-updaters running during rollout

**Describe the bug**

It is expressed in the documentation to not run more than one image-updater (per application, at least), but I notice when doing a rollout (eg: for a config or secret change) the old pod stays up until the new one is up. I wonder if that can cause problems under certain conditions?

![image](https://user-images.githubusercontent.com/10092581/150222772-033d3504-7010-4b84-a5c3-3a537bad79a9.png)

**To Reproduce**

`kubectl -n argocd rollout restart deployment argocd-image-updater`

**Expected behavior**

Given the guidance in the documentation about concurrent instances, I would expect no more than one replica running at a given time, and in the case of a failed deployment, rollback is done.

**Additional context**

https://github.com/argoproj-labs/argocd-image-updater/blob/7ed1f675ca2373dd8f0442e1a398e3fc9a227c55/manifests/base/deployment/argocd-image-updater-deployment.yaml#L1-L8

**Version**

```
argocd-image-updater: v0.11.3+f62b7d7
  BuildDate: 2022-01-19T22:16:42Z
  GitCommit: f62b7d7841a42030fe4c75de1951c46b0e7fca2e
  GoVersion: go1.16.13
  GoCompiler: gc
  Platform: linux/amd64
```

**Logs**

N/A

Thanks @jannfis 

I've run `make manifests` but I get the error below (because I don't have `kustomize2`:

```
MAGE_NAMESPACE=quay.io/argoprojlabs ./hack/generate-manifests.sh
+++ dirname ./hack/generate-manifests.sh
++ CDPATH=
++ cd -- ./hack/..
++ pwd -P
+ SRCROOT=/Users/nick/dev/github.com/NickLarsenNZ/argocd-image-updater
+ KUSTOMIZE=kustomize2
++ mktemp /tmp/aic-manifests.XXXXXX
+ TEMPFILE=/tmp/aic-manifests.qELwg8
+ IMAGE_NAMESPACE=quay.io/argoprojlabs
+ IMAGE_TAG=
+ '[' '' = '' ']'
++ git rev-parse --abbrev-ref HEAD
+ branch=update-strategy-recreate
+ [[ update-strategy-recreate = release-* ]]
+ '[' '' = '' ']'
+ IMAGE_TAG=latest
+ cd /Users/nick/dev/github.com/NickLarsenNZ/argocd-image-updater/manifests/base
+ kustomize2 edit set image quay.io/argoprojlabs/argocd-image-updater:latest
./hack/generate-manifests.sh: line 28: kustomize2: command not found
make: *** [manifests] Error 127
```

I have also tried `KUSTOMIZE='kubectl kustomize' make manifests` which fails with:

```
IMAGE_NAMESPACE=quay.io/argoprojlabs ./hack/generate-manifests.sh
+++ dirname ./hack/generate-manifests.sh
++ CDPATH=
++ cd -- ./hack/..
++ pwd -P
+ SRCROOT=/Users/nick/dev/github.com/NickLarsenNZ/argocd-image-updater
+ KUSTOMIZE='kubectl kustomize'
++ mktemp /tmp/aic-manifests.XXXXXX
+ TEMPFILE=/tmp/aic-manifests.QVZLE6
+ IMAGE_NAMESPACE=quay.io/argoprojlabs
+ IMAGE_TAG=
+ '[' '' = '' ']'
++ git rev-parse --abbrev-ref HEAD
+ branch=update-strategy-recreate
+ [[ update-strategy-recreate = release-* ]]
+ '[' '' = '' ']'
+ IMAGE_TAG=latest
+ cd /Users/nick/dev/github.com/NickLarsenNZ/argocd-image-updater/manifests/base
+ kubectl kustomize edit set image quay.io/argoprojlabs/argocd-image-updater:latest
Error: specify one path to kustomization.yaml
make: *** [manifests] Error 1
```

I'm happy to install `kustomize2`, but I hadn't heard of it. Is this it: https://github.com/phanimarupaka/kustomize2 ?

--- a/manifests/base/deployment/argocd-image-updater-deployment.yaml
+++ b/manifests/base/deployment/argocd-image-updater-deployment.yaml
@@ -10,6 +10,8 @@ spec:
   selector:
    matchLabels:
     app.kubernetes.io/name: argocd-image-updater
+  strategy:
+    type: Recreate
   template:
     metadata:
       labels:

--- a/manifests/install.yaml
+++ b/manifests/install.yaml
@@ -85,6 +85,8 @@ spec:
   selector:
     matchLabels:
       app.kubernetes.io/name: argocd-image-updater
+  strategy:
+    type: Recreate
   template:
     metadata:
       labels:

